ansible/roles-infra/infra-osp-template-create/tasks/create_osp_template.yml

```diff
@@ -1,69 +1,100 @@
-
   block:
-    - name: Check if Heat stack already exists
-      command: >-
-        openstack stack show -f json base-stack-{{ guid }}
-      register: r_show
-      failed_when: false
-      changed_when: false
-      tags: heat
-
-    - when: r_show.rc == 0
-      block:
-
-        - name: Update Heat stack from generated template
-          command: >-
-            openstack stack update -f json --wait
-            -t {{ heat_master_template }}
-            base-stack-{{ guid }}
-          register: r_hot_out_update
-
-    - when: r_show.rc != 0
-      block:
+  - name: Update Heat stack from generated template
+    when: r_stack_list.stdout | from_json | json_query(stack_name)
+    command: >-
+      openstack stack update -f json --wait
+      -t {{ heat_master_template }}
+      base-stack-{{ guid }}
+    vars:
+      stack_name: >-
+        [?"Stack Name"=='base-stack-{{ guid }}']
+    register: r_hot_out_update
 
-        - name: Create Heat stack from generated template
-          command: >-
-            openstack stack create -f json --wait
-            -t {{ heat_master_template }}
-            base-stack-{{ guid }}
-          register: r_hot_out_create
+  - name: Create Heat stack from generated template
+    when: not r_stack_list.stdout | from_json | json_query(stack_name)
+    command: >-
+      openstack stack create -f json --wait
+      -t {{ heat_master_template }}
+      base-stack-{{ guid }}
+    vars:
+      stack_name: >-
+        [?"Stack Name"=='base-stack-{{ guid }}']
+    register: r_hot_out_create
 
-    - name: Grab outputs
-      command: >-
-        openstack stack show -f json base-stack-{{ guid }}
-      changed_when: false
-      register: r_hot_outputs
+  - name: Grab outputs
+    command: >-
+      openstack stack show -f json base-stack-{{ guid }}
+    changed_when: false
+    register: r_hot_outputs
 
-    - debug:
-        var: r_hot_outputs
-        verbosity: 3
+  - debug:
+      var: r_hot_outputs
+      verbosity: 3
 
-    - set_fact:
-        hot_outputs: "{{ (r_hot_outputs.stdout | from_json) }}"
-    
-    - debug:
-        var: hot_outputs
-        verbosity: 3
+  - set_fact:
+      hot_outputs: "{{ (r_hot_outputs.stdout | from_json) }}"
+  
+  - debug:
+      var: hot_outputs
+      verbosity: 3
 
-    - name: Save infra_key content
-      set_fact:
-        infra_private_key_content: "{{ hot_outputs | json_query(query) }}"
-      vars:
-        query: "outputs[?@.output_key=='{{ guid }}-infra_key'].output_value|[0]"
+  - name: Save infra_key content
+    set_fact:
+      infra_private_key_content: "{{ hot_outputs | json_query(query) }}"
+    vars:
+      query: "outputs[?@.output_key=='{{ guid }}-infra_key'].output_value|[0]"
 
-    - name: Set infra_ssh_key path
-      set_fact:
-        infra_ssh_key: "{{ output_dir }}/{{ guid }}_infra_ssh_key.pem"
+  - name: Set infra_ssh_key path
+    set_fact:
+      infra_ssh_key: "{{ output_dir }}/{{ guid }}_infra_ssh_key.pem"
```

Comment:
These `set_fact` tasks should likely be combined?

Reply:
I'm not touching things that are not in the scope of this PR. This can be fixed separately when someone has more time to test...for now, lets please focus on the feature being added.